src/Client.php

```diff
@@ -284,158 +169,128 @@ public function captureLastError(array $payload = [])
-
-        if (null === $this->lastEvent) {
-            return null;
-        }
-
-        return str_replace('-', '', $this->lastEvent->getId()->toString());
+        return $this->transport->send($event);
     }
 
     /**
-     * {@inheritdoc}
+     * Assembles an event and prepares it to be sent of to Sentry.
+     *
+     * @param array      $payload the payload that will be converted to an Event
+     * @param null|Scope $scope   optional scope which enriches the Event
+     *
+     * @return null|Event returns ready to send Event, however depending on options it can be discarded
      */
-    public function capture(array $payload)
+    private function prepareEvent(array $payload, ?Scope $scope = null): ?Event
     {
-        $event = new Event($this->config);
+        $sampleRate = $this->getOptions()->getSampleRate();
+        if ($sampleRate < 1 && mt_rand(1, 100) / 100.0 > $sampleRate) {
+            return null;
+        }
+
+        $event = new Event();
+        $event->setServerName($this->getOptions()->getServerName());
+        $event->setRelease($this->getOptions()->getRelease());
+        $event->setEnvironment($this->getOptions()->getCurrentEnvironment());
+        $event->getTagsContext()->merge($this->getOptions()->getTags());
 
         if (isset($payload['transaction'])) {
             $event->setTransaction($payload['transaction']);
         } else {
-            $event->setTransaction($this->transactionStack->peek());
+            $request = ServerRequestFactory::fromGlobals();
+            $serverParams = $request->getServerParams();
+
+            if (isset($serverParams['PATH_INFO'])) {
+                $event->setTransaction($serverParams['PATH_INFO']);
+            }
         }
 
         if (isset($payload['logger'])) {
             $event->setLogger($payload['logger']);
         }
 
-        $event = $this->middlewareStack->executeStack(
-            $event,
-            isset($_SERVER['REQUEST_METHOD']) && \PHP_SAPI !== 'cli' ? ServerRequestFactory::fromGlobals() : null,
-            isset($payload['exception']) ? $payload['exception'] : null,
-            $payload
-        );
+        $message = $payload['message'] ?? null;
+        $messageParams = $payload['message_params'] ?? [];
 
-        $this->send($event);
-
-        $this->lastEvent = $event;
-
-        return str_replace('-', '', $event->getId()->toString());
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function send(Event $event)
-    {
-        if (!$this->config->shouldCapture($event)) {
-            return;
+        if (null !== $message) {
+            $event->setMessage(substr($message, 0, self::MESSAGE_MAX_LENGTH_LIMIT), $messageParams);
         }
 
-        // should this event be sampled?
-        if (mt_rand(1, 100) / 100.0 > $this->config->getSampleRate()) {
-            return;
+        if (isset($payload['exception'])) {
```

Comment:
You're assuming this contains a `\Throwable`. Can we assume that?

Reply:
I've added a check in `addThrowableToEvent`

`if (null === $exception || !$exception instanceof \Throwable) {`